packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -430,11 +433,16 @@ export class SummaryGenerator {
 			};
 			if (ackNackOp.type === MessageType.SummaryAck) {
 				this.heuristicData.markLastAttemptAsSuccessful();
-				this.successfulSummaryCallback();
 				summarizeEvent.end({
 					...summarizeTelemetryProps,
 					handle: ackNackOp.contents.handle,
 				});
+				await this.successfulSummaryCallback({
```

Comment:
This is where refresh is called

Reply:
This is fine. SummaryGenerator only cares about summarization and not about refreshing state from an ack.

The only small issue I see here is that if this callback isn't awaited, then things will not work. I guess that's true for a lot of other logic so it seems fine.

Reply:
There was a test failing, so I split it into two.